services/orderService.js

```diff
@@ -0,0 +1,80 @@
+const orderDao = require('../models/orderDao');
+const orderStatusEnums = require('../models/enums');
+
+const createOrder = async ( userId, roomId, checkIn, checkOut ) => {
+  
+  const getRoomPriceByRoomId = await orderDao.getRoomPriceByRoomId( +roomId );
+  
+  const getUserPointByUserId = await orderDao.getUserPointByUserId( +userId );
+ 
+
+  
+  let pointChange = Number(getUserPointByUserId[0].point) - Number(getRoomPriceByRoomId[0].price);
+
+if( +getUserPointByUserId === 0 ){
+  const err = new Error('USER_POINT_IS_EMPTY');
+  err.statusCode = 400;
+  throw err;
+}
+
+if( pointChange < 0){
+  const err = new Error('PLEASE_ADD_YOUR_POINT');
+  err.statusCode = 400;
+  throw err;
+}
+
+if( pointChange > 0 ) {
+
+
+  const orderData = await orderDao.createOrder( +userId , +roomId, checkIn, checkOut , pointChange, orderStatusEnums.RESERVATE_ON_ORDER_STATUS_ID );
+  
+  
+  return orderData;
+}
```

Comment:
현태님 함수 내부에서 작동하는 조건문이기 떄문에 들여쓰기 신경써주세요.

첫번째 조건문은 특정 값이 0인 경우로서 아래와 같이 작성함으로써 가독성을 높일 수 있습니다.

```suggestion
  if(  !getUserPointByUserId ){
    const err = new Error('USER_POINT_IS_EMPTY');
    err.statusCode = 400;
    throw err;
  }

  if( pointChange < 0){
    const err = new Error('PLEASE_ADD_YOUR_POINT');
    err.statusCode = 400;
    throw err;
  }

  if( pointChange > 0 ) {
  const orderData = await orderDao.createOrder( +userId , +roomId, checkIn, checkOut , pointChange, orderStatusEnums.RESERVATE_ON_ORDER_STATUS_ID ); 
  
  return orderData;
}
```